dtfs-central-api/src/v1/validation/utilisation-report-service/fee-record-validator.ts

```diff
@@ -0,0 +1,53 @@
+import { FEE_RECORD_STATUS, FeeRecordEntity, FeeRecordStatus } from '@ukef/dtfs2-common';
+import { FeeRecordRepo } from '../../../repositories/fee-record-repo';
+import { InvalidPayloadError } from '../../../errors';
+
+/**
+ * Validates that all provided fee records have the same payment currency.
+ * If there is only one fee record or less, validation passes.
+ * @param feeRecords - The fee records to validate
+ * @throws {InvalidPayloadError} If fee records have different payment currencies
+ */
+export const validateFeeRecordsAllHaveSamePaymentCurrency = (feeRecords: FeeRecordEntity[]) => {
+  if (feeRecords.length <= 1) {
+    return;
+  }
+  const { paymentCurrency } = feeRecords[0];
+  if (feeRecords.some((feeRecord) => feeRecord.paymentCurrency !== paymentCurrency)) {
+    throw new InvalidPayloadError('Fee records must all have the same payment currency');
+  }
+};
+
+/**
+ * Validates that all provided fee records with payments have the same fee
+ * record payment group and that the fee record IDs match the fee record IDs on
+ * this existing payment group.
+ * @param feeRecordIds - The fee record IDs
+ * @throws {InvalidPayloadError} If fee records have different statuses
+ * @throws {InvalidPayloadError} If fee record IDs do not match the fee record IDs on the existing payment group
+ */
+export const validateFeeRecordsWithPaymentsAreOnePaymentGroup = async (feeRecordIds: number[]) => {
```

Comment:
I'm not entirely convinced on this name since what we actually are doing is:
-> do they all match in status and that status is not DOES_NOT_MATCH? -> fine
-> do they all match, status is DOES_NOT_MATCH, and they form a complete group? -> fine
-> otherwise -> not fine.
So e.g. I could pass in three fee records with payments that are in Match status but not the fourth payment from their group and this would say 'fine' which is not what the function name states at all.

I think we want a mix of logic change and a mix of name change. Ideally:
-> do they all match in status and that status is TO_DO? -> fine (we are forming a new group)
-> do they all match in status and they form a complete group? -> fine
-> otherwise -> not fine.
And then for naming perhaps: `validateFeeRecordsFormCompleteGroup` or if we care to be explicit about group formation can be `validateFeeRecordsFromCompleteGroupOrHaveNoPayments` but that is perhaps too long so I think perhaps fine to leave off.

Reply:
Made this logic ordering change and updated the tests, also updated the name thanks